dumper/app/src/test/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/ClouderaClusterCPUChartTaskTest.java

```diff
@@ -0,0 +1,140 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper.connector.cloudera.manager;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyChar;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.io.ByteSink;
+import com.google.common.io.CharSink;
+import com.google.edwmigration.dumper.application.dumper.connector.cloudera.manager.ClouderaManagerHandle.ClouderaClusterDTO;
+import com.google.edwmigration.dumper.application.dumper.task.TaskRunContext;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.Writer;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashSet;
+import java.util.Set;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ClouderaClusterCPUChartTaskTest {
+  private final ClouderaClusterCPUChartTask task = new ClouderaClusterCPUChartTask();
+  private ClouderaManagerHandle handle;
+  private String servicesJson;
+
+  @Mock private TaskRunContext context;
+  @Mock private ByteSink sink;
+  @Mock private Writer writer;
+  @Mock private CharSink charSink;
+  @Mock private CloseableHttpClient httpClient;
+  private URI uri;
+
+  @Before
+  public void setUp() throws Exception {
+    servicesJson = readFileAsString("/cloudera/manager/cluster-cpu-status.json");
+    uri = URI.create("http://localhost/api");
+    handle = new ClouderaManagerHandle(uri, httpClient);
+
+    when(sink.asCharSink(eq(StandardCharsets.UTF_8))).thenReturn(charSink);
+    when(charSink.openBufferedStream()).thenReturn(writer);
+  }
+
+  @Test
+  public void noClusterId_skipWrites() throws Exception {
+    // GIVEN: There is no cluster with a valid cluster ID
+    handle.initClusters(ImmutableList.of(ClouderaClusterDTO.create(null, "single cluster")));
```

Comment:
Let's add test for at least 2 clusters, to maker test that task do requests for all clusters.

Reply:
Added.